tests/testsuite/build.rs

```diff
@@ -1563,6 +1578,208 @@ fn crate_env_vars() {
     }
 }
 
+#[cargo_test]
+fn crate_env_vars_with_workspace() {
```

Comment:
Thank you for setting up this new test case. Given it is largely the same as `crate_env_vars_without_workspace`, should we test the part we need (`CARGO_WORKSPACE_DIR`) instead of duplicating it?

Or alternatively we could find a way to avoid the copy.